website/content/docs/connect/manage-traffic/route-to-local-upstreams.mdx

```diff
@@ -83,23 +83,252 @@ You can configure the default routing behavior for all proxies in the mesh as we
-1. Apply the configuration by either calling the [`/config` HTTP API endpoint](/consul/api-docs/config) or running the [`consul config write` CLI command](/consul/commands/config/write). 
-   The following example writes a proxy defaults configuration entry from a local HCL file using the CLI:
+Create a service resolver configuration entry and specify the following fields:
+  - `Name`: The name of the target upstream service for which downstream clients should use locality-aware routing.
+  - `PrioritizeByLocality`: This block enables proxies in the mesh to use the region and zone defined in the service configuration to route traffic. Set the `mode` inside the block to `failover`. Refer to [`PrioritizeByLocality`](/consul/docs/connect/config-entries/service-resolver#prioritizebylocality) in the service resolver reference for details about the configuration.
+
+<Tabs>
+<Tab heading="HCL" group="hcl">
+<CodeBlockConfig filename="api-resolver.hcl">
 
-   ```shell-session
-   $ consul config write web-resolver.hcl
-   ```
+```hcl
+Kind     = "service-resolver"
+Name     = "api"
+PrioritizeByLocality = {
+    Mode = "failover"
+}
+```
+</CodeBlockConfig>
+</Tab>
+
+<Tab heading="JSON" group="json">
+<CodeBlockConfig filename="api-resolver.json">
+
+```json
+{
+  "kind": "service-resolver",
+  "name": "api",
+  "prioritizeByLocality": {
+    "mode": "failover"
+  }
+}
+```
+</CodeBlockConfig>
+</Tab>
+<Tab heading="Kubernetes" group="kubernetes">
+<CodeBlockConfig filename="api-resolver.yaml">
+
+```yaml
+apiversion: consul.hashicorp.com/v1alpha1
+kind: ServiceResolver
+metadata:
+  name: api
+spec:
+  prioritizeByLocality:
+    mode: failover
+```
+
+</CodeBlockConfig>
+</Tab>
+</Tabs>
+
+Apply the configuration by either running the [`consul config write` CLI command](/consul/commands/config/write), applying the Kubernetes CRD, or calling the [`/config` HTTP API endpoint](/consul/api-docs/config).
+
+<Tabs>
+<Tab heading="Consul CLI" group="cli">
+
+  ```shell-session
+  $ consul config write api-resolver.hcl
+  ```
+
+</Tab>
+
+<Tab heading="Kubernetes" group="k8s">
+
+  ```shell-session
+  $ kubectl apply -f api-resolver.yaml
+  ```
+
+</Tab>
+<Tab heading="HTTP API" group="api">
+
+  ```shell-session
+  $ curl --request PUT --data @api-resolver.hcl http://127.0.0.1:8500/v1/config
+  ```
+
+</Tab>
+</Tabs>
+
+### Explicit locality configuration on Kubernetes test clusters
+
+If you are testing this feature with a local Kubernetes cluster (or in an environment where `topology.kubernetes.io` labels are not set), it is possible to explicitly provide each node's locality using `kubectl label`:
+
+```shell-session
+kubectl label node $K8S_NODE topology.kubernetes.io/region="us-east-1" topology.kubernetes.io/zone="us-east-1a"
+```
+
+After setting these values, you should see subsequent service and proxy registrations in your cluster inherit the values from their local Kubernetes node.
```

Comment:
```suggestion
### Configure locality on Kubernetes test clusters explicitly

You can explicitly configure locality for each Kubernetes node If you are testing locality-aware routing with a local Kubernetes cluster or in an environment where `topology.kubernetes.io` labels are not set. Run the `kubectl label node` command specify the locality as arguments. The following example specifies the `us-east-1` region and `us-east-1a` zone for the node:

```shell-session
kubectl label node $K8S_NODE topology.kubernetes.io/region="us-east-1" topology.kubernetes.io/zone="us-east-1a"
```

After setting these values, subsequent service and proxy registrations in your cluster inherit the values from their local Kubernetes node.
```
I was trying to keep the consistent headings to start with "Configure . . .", but change it back if we need to emphasize that this is an explicit operation.